llama-index-integrations/vector_stores/llama-index-vector-stores-weaviate/llama_index/vector_stores/weaviate/base.py

```diff
@@ -383,3 +477,40 @@ def query(self, query: VectorStoreQuery, **kwargs: Any) -> VectorStoreQueryResul
         return VectorStoreQueryResult(
             nodes=nodes, ids=node_ids, similarities=similarities
         )
+
+    def query(self, query: VectorStoreQuery, **kwargs: Any) -> VectorStoreQueryResult:
+        """Query index for top k most similar nodes."""
+        collection = self._client.collections.get(self.index_name)
```

Comment:
`self._client` might be `None` at this point. You might want to isolate the logic `if self._client is not None...` in a private property or method to avoid repeating it over and over, something like:
```
@property
def client(self):
    if self._client is None:
        raise ClientNotProvided(...)
    return self._client

@property
def async_client(self):
    if self._aclient is None:
        raise AsyncClientNotProvidedError(...)
    return self._aclient
```

Reply:
Implemented in https://github.com/run-llama/llama_index/pull/17220/commits/eed533a48719d8742da90c607e0b6296d023bff2